From 6e28915d5cdf6b0b430e9de1369652408c4cd739 Mon Sep 17 00:00:00 2001
From: Tim B <timb@antifact.com>
Date: Tue, 29 Oct 2013 14:22:59 -0700
Subject: [PATCH] Issue #2122535 by timb: Fixed all instances of conditions
 misspelling.

---
 drupagram.inc | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drupagram.inc b/drupagram.inc
index a2d397b..f78fa81 100644
--- a/drupagram.inc
+++ b/drupagram.inc
@@ -207,18 +207,18 @@ function drupagram_account_delete($drupagram_id) {
 
   // Delete from {drupagram_account}.
   $query = db_delete('drupagram_account');
-  $query->conditions('drupagram_id', $drupagram_id);
+  $query->condition('drupagram_id', $drupagram_id);
   $query->execute();
 
   // Delete from {drupagram}.
   $query = db_delete('drupagram');
-  $query->conditions('username', $account->username);
+  $query->condition('username', $account->username);
   $query->execute();
 
   // Delete from {drupagram_account}.
   $query = db_delete('authmap');
-  $query->conditions('authname', $drupagram_id);
-  $query->conditions('module', 'drupagram');
+  $query->condition('authname', $drupagram_id);
+  $query->condition('module', 'drupagram');
   $query->execute();
 }
 
-- 
1.8.3.4 (Apple Git-47)

